docs/getting-started/basics.md

```diff
@@ -1,7 +1,150 @@
+structure and properties of these graph elements are defined in the elements schema. The general
+format of an element schema are two lists; one of the `"edges"` and the other of the `"entities"`
+like the following:
+
+!!! example "Basic elements syntax"
+
+    ```json
+    {
+        "edges": {
+            "Edge": {
+                "source": "type",
+                "destination": "type",
+                "directed": "true",
+                "properties": {
+                    "property": "type"
+                }
+            }
+        },
+        "entities": {
+            "Node": {
+                "description": "A Node",
+                "vertex": "type",
+                "properties": {
+                    "property": "type"
+                }
+            }
+        }
+    }
+    ```
+
+As you can see there are a few fields for both the example `"Edge"` and `"Node"`, many of these
+require a type as their value (discussed in the [next section](#types-schema)) which, are
+essentially handlers or object types for the value associated with the field.
+
+For an `edge` the following fields are required.
+
+- `source` - A user defined type for the source node the edge originated from.
+
+- `directed` - Boolean true or false (this is still needs to be user defined in the types schema) to
+    define if the edge is directed or not. When an Edge is undirected in Gaffer, it is treated as if
+    the relationship was bidirectional and the vertices of the edge do not have an authoritative
+    source and destination.
+
+- `destination` - A user defined type for the destination node the edge goes to.
+
+For an entity only one field is actually required outlined, vertex (see below). The example does
+however include some of the common optional fields too such as a `"properties"` list and
+`"description"`.
+
+- `vertex` - A user defined type for the node/vertex
+
+### Types Schema
+
+Following on from the elements schema, the other necessary schema needed for a Gaffer deployment is
+the types schema.  The types schema allows user defined types for all elements in the graph. It can
+also demonstrate the power of Gaffer as it allows for custom functions classes to be used on the
+types; however, this can make it quite complex to write a full schema for a graph.
+
+!!! example "Example types syntax"
+
+    ```json
+    {
+        "types": {
+            "type.string": {
+                "description": "A basic type to hold the string value of an element",
+                "class": "java.lang.String"
+            },
+            "type.int": {
+                "description": "A basic type to hold the int value of an element",
+                "class": "java.lang.Integer"
+            }
+        }
+    }
+    ```
+
+## The Rest API
+
+Most of the interaction with a deployed Gaffer graph will be through the rest API. When deployed,
+the rest API will be available at defined address, accessing this address via a browser brings up a
```

Comment:
think you are trying to say the 'rest API will be available at a defined address', might want to just re-phrase this bit